lib/matplotlib/dates.py

```diff
@@ -1234,6 +1234,9 @@ def __init__(self, tz=None, minticks=5, maxticks=None,
             MICROSECONDLY: [1, 2, 5, 10, 20, 50, 100, 200, 500, 1000, 2000,
                             5000, 10000, 20000, 50000, 100000, 200000, 500000,
                             1000000]}
+        if interval_multiples:
+            self.intervald[DAILY] = [1, 2, 4, 7, 14, 21]
```

Comment:
Perhaps just curiosity, but why have this substitution of 4 for 3 in this case, and not in general?  And isn't 21 a rather strange option?  It is inconsistent with the docstring.

Reply:
To clarify, the 21 option is present in the current code as well as in the line cited above, so the inconsistency predates your PR.

Reply:
Yeah, I don't know why the 21 is in there.  I left it in as possibly some weird edge case, but I don't see how it'd happen.

3 vs 4?

3:   1, 4,..., 28, 31, 1
4:   1, 5, ... 25, 29, 1

So except for Feb leap years it avoids an awkward cross-month transition.

Reply:
added a couple of comments...